<?xml version="1.0"?>

<launch>
	<arg name="controller" value="fnn"/> <!-- pid, fnn -->

	<arg name="noise" default="0.00"/>

	<remap from="/uav/command_velocity_fnn" to="/y6/command_velocity"/>
	<!--remap from="/uav/command_velocity_fnn" to="/y6/command_velocity"/CHANGE fnn.launch-->
	<remap from="/uav/command" to="/y6/command"/>
	<remap from="/uav/position" to="/y6/position"/>
	<remap from="/uav/velocity" to="/y6/linear_velocity"/>
	<remap from="/uav/noise/position" to="/y6/noise/position"/>
	<remap from="/uav/odometry" to="/y6/odometry"/>
	<remap from="/uav/true_odometry" to="/y6/ground_truth/odometry"/>
	<remap from="/uav/trajectory" to="/y6/trajectory"/>
	<remap from="/uav/trajectory_velocity" to="/y6/trajectory_velocity"/>
	<remap from="/uav/noise" to="/y6/noise"/>
	<remap from="/optitrack/odometry" to="/mavros/mocap/pose"/>
	<remap from="/uav/trajectory" to="/y6/command_position"/>
	<remap from="/y6/command_attitude" to="/y6/command_velocity"/>

	<node name="safe_y6" pkg="controllers" type="safe_y6" output="screen" args=""/>

	<!-- node name="noise" pkg="controllers" type="noise" output="screen" args="$(arg noise)"/ -->
	<!-- node name="logger" pkg="controllers" type="logger" output="screen" args="$(arg controller)"/ -->

	<node name="fake_mocap" pkg="controllers" type="fake_mocap" output="screen" args="">		
		<remap from="/in_odometry" to="/ground_truth/odometry"/>
		<remap from="/out_odometry" to="/mavros/mocap/pose"/>
	</node>

	<!-- include file="$(find mocap_optitrack)/launch/mocap.launch"/ -->
	<include file="$(find controllers)/launch/$(arg controller).launch"/>

	<node name="trajectory" pkg="controllers" type="trajectory" output="screen" args="~/catkin_ws/src/controllers/policy/waypoints.txt"/>
	<node name="teleop_key" pkg="controllers" type="teleop_key" output="screen" args=""/>
	<node name="motor_failure_pub" pkg="controllers" type="motor_failure_pub" output="screen" args=""/>
	<node name="dynamic_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" args=""/>
</launch>
